Update documentation on retention policies limits (#8529)
matrix-org/synapse: docs/message_retention_policies.md

### Lifetime limits

Server admins can set limits on the values of `max_lifetime` to use when
purging old events in a room. These limits can be defined as such in the
`retention` section of the configuration file:

```yaml
  allowed_lifetime_min: 1d
  allowed_lifetime_max: 1y
```

> These values are evaluated when running purge jobs, to adjust the

Reviewer: ```suggestion
The limits are considered when running purge jobs, to adjust the
```

> effective value of `max_lifetime`, and make sure it's within the

Reviewer: ```suggestion
effective value of `max_lifetime`, to make sure it's within the
```

Author: Not sure about the "..., to ..., to ..." formulation, it looks more hard to read to me.

Reviewer: Sorry, yes that doesn't make sense. I think this sentence is a bit of a run-on though. I'd split it up with:

The limits are considered when running purge jobs. If necessary, the value of `max_lifetime` will be brought within the `[allowed_lifetime_min, allowed_lifetime_max]` range.

Author: Fixed in https://github.com/matrix-org/synapse/pull/8529/commits/ddd1f2bc9af1130e5ce209eb44f4c3c168115f19

> `[allowed_lifetime_min, allowed_lifetime_max]` range. If the value of

Reviewer: ```suggestion
`[allowed_lifetime_min, allowed_lifetime_max]` range, inclusive. If the value of
```

Author: The writing `[a, b]` already imply it's inclusive. An exclusive version would look like `]a, b[` (fully exclusive) or `[a, b[` (partially exclusive). See https://en.wikipedia.org/wiki/Interval_(mathematics)#Including_or_excluding_endpoints

Reviewer: I know, but server admins may not understand mathematical notation is all.

Author: I'll probably rephrase it so it uses plain English instead of a weird mix of both then.

Author: Done in https://github.com/matrix-org/synapse/pull/8529/commits/ddd1f2bc9af1130e5ce209eb44f4c3c168115f19
